docs/docs/20-concepts/02-mcp-gateway/03-obot-registry.md

```diff
@@ -0,0 +1,288 @@
+env:
+  - key: API_KEY                     *# Actual env var name*
+    name: API Key                    *# Display name in UI*
+    required: true                   *# Is this mandatory?*
+    sensitive: true                  *# Masks value in UI (for credentials)*
+    description: Your API key from the developer dashboard
+
+*# Preview your server's tools (list top 20 if you have many)*
+toolPreview:
+  - name: search_documents
+    description: Search through indexed documents using natural language
+    params:
+      query: Search query string
+      limit: Maximum number of results (1-100)
+  - name: get_document
+    description: Retrieve a specific document by ID
+    params:
+      document_id: Unique identifier for the document
+```
+
+Now choose **exactly one** runtime block below.
+
+<details>
+  <summary>Option A: Remote HTTP Server</summary>
+
+Choose the case that matches your setup:
+
+### Case 1: Fixed Endpoint URL
+
+Use this when your server has one static URL for all users:
+
+```yaml
+runtime: remote
+remoteConfig:
+  fixedURL: https://api.example.com/v1/mcp
+  headers:
+    - name: Personal Access Token
+      description: PAT
+      key: Authorization           # HTTP header name
+      required: true
+      sensitive: true
+
+```
+
+### Case 2: Same Hostname, User Selects Path
+
+Use this when users connect to different paths on your domain:
+
+```yaml
+runtime: remote
+remoteConfig:
+  hostname: api.example.com
+  headers:
+    - name: API Key
+      description: Your API key from <https://example.com/settings>
+      key: X-API-Key
+      required: true
+      sensitive: true
+
+```
+
+Users will specify their path when connecting (e.g., https://api.example.com/serviceA/mcp).
+
+### Case 3: URL Built from User Environment
+
+Use this when the URL includes user-specific values:
+
+```yaml
+env:
+  - key: WORKSPACE_URL
+    name: Workspace URL
+    description: "Your workspace URL, e.g., <https://mycompany.cloud.com>"
+    required: true
+    sensitive: false
+
+runtime: remote
+remoteConfig:
+  URLTemplate: ${WORKSPACE_URL}/api/2.0/mcp/
+  headers:
+    - name: Personal Access Token
+      description: PAT with workspace access
+      key: Authorization
+      required: true
+      sensitive: true
+
+```
+
+---
+</details>
+
+<details>
+  <summary>Option B: Containerized HTTP Server</summary>
+
+Use this if you already have a Docker image that serves streamable HTTP:
+
+```yaml
+runtime: containerized
+containerizedConfig:
+  image: your-dockerhub-user/your-mcp-server:v1.0.0
+  port: 8080           # Your container's exposed HTTP port
+  path: /mcp           # HTTP path where MCP endpoint is served
+  args:                # Optional runtime flags
+    - flags # flags needed.
+
+```
+
+**Requirements:**
+
+- Your container must serve HTTP/SSE on the specified port
+- The MCP endpoint must be available at the specified path
+- Image must be publicly accessible (Docker Hub, GHCR, etc.)
```

Comment:
Instead of calling this out, I would just say something about how the image must be repackaged in mcp-images

Reply:
reworked, added instructions to repackage http docker image.